lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -3409,6 +3449,41 @@ ParseResult FIRStmtParser::parseInstance() {
   return moduleContext.addSymbolEntry(id, entryId, startTok.getLoc());
 }
 
+/// object ::= 'object' id 'of' id info?
+ParseResult FIRStmtParser::parseObject() {
+  if (FIRVersion::compare(version, {3, 1, 0}) < 0)
+    return emitError() << "unexpected token: objects are a FIRRTL 3.1.0+ "
+                          "feature, but the specified FIRRTL version was "
+                       << version;
+
+  auto startTok = consumeToken(FIRToken::kw_object);
+
+  // If this was actually the start of a connect or something else handle
+  // that.
+  if (auto isExpr = parseExpWithLeadingKeyword(startTok))
+    return *isExpr;
+
+  StringRef id;
+  StringRef className;
+  if (parseId(id, "expected object name") ||
+      parseToken(FIRToken::kw_of, "expected 'of' in object") ||
+      parseId(className, "expected class name") || parseOptionalInfo())
```

Comment:
I know it matches instance instantiation like this:

```
object foo of Foo
```

But what do you think about doing it like this:

```
object foo of Inst<Foo>
```

Having the type of an object spelled differently in ports (`output foo: Inst<Foo>`) versus instantiations (`object foo of Foo`, no `Inst<>`), means we need to differentiate how we serialize these into FIRRTL depending on the context, when conceptually they are the same type: the type of an instance of Foo.

If we use the `Inst<>` here, or remove it in the other context, it seems like a simpler IR, since we always spell the type of an instance the same. What do you think?

Reply:
Alright, I rescind my previous comment, it makes sense to do this the same as module instantiation for the reasons you outlined offline. I was confusing myself because I'd written the Chisel serializer for this construct in a weird way; rewriting it to work the same as the Chisel serializer for module instantiation made it clear this is straightforward.